Picture.py

```diff
@@ -10,12 +10,33 @@
 
 class Picture:
 
-    def __init__(self, image=None, extension=".jpg"):
-        self.image = image
-        self.extension = extension
-        try:
-            self.filename = self.title = image.filename
-        except AttributeError:
+    def __init__(self, arg=None, extension=".jpg"):
+        if isinstance(arg, Picture):
+            # arg is a Picture
+            self.image = arg.image.copy()
+            self.extension = arg.extension
+            self.filename = arg.filename
+            self.title = arg.title
+        elif isinstance(arg, PIL.Image.Image):
+            # arg is a PIL image
+            self.image = arg
+            self.extension = extension
+            try:
+                self.filename = self.title = arg.filename
+            except AttributeError:
+                self.filename = self.title = ''
+        elif isinstance(arg, str):
+            # arg is a string, do what JES does and make picture of string
+            self.image = PIL.Image.new("RGB", (600, 200))
+            self.extension = extension
+            self.title = arg
+            self.filename = ''
+            draw = PIL.ImageDraw.Draw(self.image)
```

Comment:
Why not just use self.addText() rather than drawing the text using PIL draw

Reply:
Given that this is happening in the constructor (init function?) for the class, I don't think we have "self" defined yet.